core/src/iam/signin.rs

```diff
@@ -129,19 +129,18 @@ pub async fn sc(
 								// Create the authentication key
 								let key = EncodingKey::from_secret(sv.code.as_ref());
 								// Create the authentication claim
+								let exp = Some(
+									match sv.session {
+										Some(v) => Utc::now() + Duration::from_std(v.0).unwrap(),
```

Comment:
I actually just moved this code (which was already there) outside of the `Claims` object, but you are right in pointing this out. Nice catch! I assume this was implemented in this way because to trigger the panic an attacker would need to be able to issue tokens signed with a credential that is trusted by SurrealDB. However, tokens can currently be issued for specific scopes by actors that are only trusted within that scope, database, namespace... Since the function is already able to return an error, I think it is much better to take this into account and to take this opportunity to check this properly.

Reply:
Resolved in [ac4186f](https://github.com/surrealdb/surrealdb/pull/3561/commits/ac4186f53ff38f420fe7dfa2cad3da85b09507b1).